frontend/src/citizen-frontend/child-documents/vasu/sections/CitizenVasuEvents.tsx

```diff
@@ -0,0 +1,117 @@
+// SPDX-FileCopyrightText: 2017-2022 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useMemo } from 'react'
+import styled from 'styled-components'
+
+import { VasuDocument } from 'lib-common/generated/api-types/vasu'
+import LocalDate from 'lib-common/local-date'
+import HorizontalLine from 'lib-components/atoms/HorizontalLine'
+import { ContentArea } from 'lib-components/layout/Container'
+import ListGrid from 'lib-components/layout/ListGrid'
+import { Dimmed, H2, Label } from 'lib-components/typography'
+import { defaultMargins } from 'lib-components/white-space'
+
+import { useTranslation } from '../../../localization'
+import { VasuStateChip } from '../components/VasuStateChip'
+import { isDateQuestion } from '../vasu-content'
+import { getLastPublished } from '../vasu-events'
+
+const labelWidth = '320px'
+
+const Container = styled(ContentArea)`
+  padding: ${defaultMargins.L};
+  ${H2} {
+    margin-top: 0;
+  }
+`
+
+const ChipContainer = styled.div`
+  display: inline-flex;
+`
+
+function EventRow({ label, date }: { label: string; date: LocalDate | null }) {
+  const i18n = useTranslation()
+
+  return (
+    <>
+      <Label>{label}</Label>
+      {date ? (
+        <span>{date.format()}</span>
+      ) : (
+        <Dimmed>{i18n.vasu.noRecord}</Dimmed>
+      )}
+    </>
+  )
+}
+
+interface Props {
+  document: Pick<
+    VasuDocument,
+    'documentState' | 'events' | 'modifiedAt' | 'type'
+  >
+  content: VasuDocument['content']
+}
+
+export function CitizenVasuEvents({
+  document: { documentState, events, modifiedAt, type },
+  content
+}: Props) {
+  const i18n = useTranslation()
+  const lastPublished = getLastPublished(events)
+
+  const trackedDates: [string, LocalDate][] = useMemo(
+    () =>
+      content.sections.flatMap((section) =>
+        section.questions
+          .filter(isDateQuestion)
+          .filter((question) => question.trackedInEvents)
+          .map(({ name, nameInEvents, value }): [string, LocalDate | null] => [
+            nameInEvents || name,
+            value
+          ])
+          .filter((pair): pair is [string, LocalDate] => pair[1] !== null)
+      ),
```

Comment:
This events section will most likely be changed after beta testing as it is missing from the UX because the part of the page it is on is not scrolled into view. Will revise it in the next PR with more understanding what should the citizen see here